deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkUri.java

```diff
@@ -1215,7 +1217,22 @@ private static boolean decodeIpv4Suffix(
      */
     private static String domainToAscii(String input) {
       try {
-        String result = IDN.toASCII(input).toLowerCase(Locale.US);
+        // When this is a template URI it might get to long with all the placeholder definitions
+        // to pass verification.
+        // This will replace the placeholders with other shorter placeholders for URI validation.
+        Set<String> placeholders = getPlaceHolders(input);
```

Comment:
this could probably be pulled out to a standalone function that could encapsulate its functionality and be more directly testable

Reply:
In Kotlin yes, in Java it is a bit tricky as there would be multiple returns if I don't create a wrapper. I'll leave this like this for now but if we ever convert this to Kotlin it makes sense.